StudentsGrades/Data/GradeItem.cs

```diff
@@ -0,0 +1,19 @@
+﻿using System.ComponentModel.DataAnnotations;
+
+namespace StudentsGrades.Data
+{
+    public class GradeItem
+    {
+        public int GradeItemId { get; set; }
+
+        [Required(ErrorMessage = "Имя не должно быть пустым!")]
+        public string Name { get; set; } = "";
+
+        [Required(ErrorMessage = "Название предмета не должно быть пустым!")]
+        public string Subject { get; set; } = "";
+
+        [Required(ErrorMessage = "Оценка должна быть выставлена!")]
+        [Range(0, 10, ErrorMessage = "Оценка должна быть числом от 0 до 10!")]
+        public int Grade { get; set; }
```

Comment:
```suggestion
        public int? Grade { get; set; }
```
Для типа значений атрибут Required не нужен, они автоматически проверяются на непустоту. Однако и выводимым сообщением в случае пустой строки управлять нельзя.

![image](https://user-images.githubusercontent.com/3691630/209587322-9558fd03-b260-44d6-9cf7-d717d9cdabad.png)

Наилучшее решение, которое я нашел - сделать тип ссылочным насильно.